### PR TITLE
📖 Docs improvement for AWS

### PR DESCRIPTION
**Pull request by randomvariable**
**What this PR does / why we need it**:
Slight cleanup for AWS docs, and some more links back to the CAPA book.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/book/src/user/quick-start.md [hunk 1]
```diff
@@ -161,10 +161,9 @@ before getting started with Cluster API. See below for the expected settings for
 {{#tabs name:"tab-installation-infrastructure" tabs:"AWS,Azure,DigitalOcean,Docker,GCP,vSphere,OpenStack,Metal3,Packet"}}
 {{#tab AWS}}
 
-Download the latest binary of `clusterawsadm` from the [AWS provider releases] and make sure to place it in your path. You need at least version v0.5.5 for these instructions.
-Instructions for older versions of clusterawsadm are available in [Github][legacy-clusterawsadm].
+Download the latest binary of `clusterawsadm` from the [AWS provider releases] and make sure to place it in your path.
 
-The clusterawsadm command line utility assists with identity and access management (IAM) for Cluster API Provider AWS.
+The [clusterawsadm][clusterawsadm] command line utility assists with identity and access management (IAM) for [Cluster API Provider AWS][capa].
```
Comment (JoelSpeed) [inline docs/book/src/user/quick-start.md:166]: Looks odd that one of these has the single `[blah]` where the others have `[blah][blah]` with the references below? Is the single `[blah]` acceptable/preferred? Should we make them consistent?

Comment (randomvariable) [reply docs/book/src/user/quick-start.md:166]: You are right. i'll fix

**Review by fabriziopandini — COMMENTED** [review]
/lgtm

**Review by vincepri — APPROVED** [review]
/approve

**Comment by sbueringer** [comment]
/retest
unrelated, known flaky test

**Pull request merged**